kernel-spark/src/test/java/io/delta/kernel/spark/Dsv2ReadTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.spark;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.spark.sql.*;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.io.TempDir;
+
+/** Tests for DSv2 batch read operations. */
+public class Dsv2ReadTest extends Dsv2TestBase {
+
+  @Test
+  public void testBatchRead() {
+    spark.sql(
+        String.format(
+            "CREATE TABLE dsv2.%s.batch_read_test (id INT, name STRING, value DOUBLE)", nameSpace));
+
+    // Select and validate the data
+    Dataset<Row> result =
+        spark.sql(String.format("SELECT * FROM dsv2.%s.batch_read_test", nameSpace));
+
+    List<Row> expectedRows = Arrays.asList();
+
+    assertDatasetEquals(result, expectedRows);
+  }
+
+  @Test
+  public void testReadWithColumnMapping(@TempDir File deltaTablePath) {
+    String tablePath = deltaTablePath.getAbsolutePath();
+
+    // Create a Delta table with column mapping enabled using name mode
+    spark.sql(
+        String.format(
+            "CREATE TABLE delta.`%s` (id INT, user_name STRING, amount DOUBLE) "
+                + "USING delta "
+                + "TBLPROPERTIES ('delta.columnMapping.mode' = 'name')",
+            tablePath));
+
+    // Insert test data
+    spark.sql(
+        String.format(
+            "INSERT INTO delta.`%s` VALUES (1, 'Alice', 100.0), (2, 'Bob', 200.0)", tablePath));
+
+    // Read through DSV2 and verify
+    Dataset<Row> result =
+        spark.sql(String.format("SELECT * FROM dsv2.delta.`%s` ORDER BY id", tablePath));
+
+    List<Row> expectedRows =
+        Arrays.asList(RowFactory.create(1, "Alice", 100.0), RowFactory.create(2, "Bob", 200.0));
+
+    assertDatasetEquals(result, expectedRows);
```

Comment:
i think you should build a better function for this .. this is verbose. ideally it should 
```
check(
     String.format("SELECT * FROM dsv2.delta.`%s` ORDER BY id", tablePath), 
     List.of( 
          List.of(1, "Alice", 100.0), 
          List.of(2, "Bob", 200.0)
     )
)
```